docs/middlewares.md

```diff
@@ -140,6 +141,104 @@ handler(event, context, (_, response) => {
 })
 ```
 
+## [functionShield](/src/middlewares/functionShield.js)
+
+Hardens AWS Lambda execution environment:
+* By monitoring (or blocking) outbound network traffic from your function, you can be certain that your data is never leaked
+* By disabling read/write operations on the /tmp/ directory, you can make your function truly ephemeral
+* By disabling the ability to launch child processes, you can make sure that no rogue processes are spawned without your knowledge by potentially malicious packages
+* By disabling the ability to read the function's (handler) source code through the file system, you can prevent handler source code leakage, which is oftentimes the first step in a serverless attack 
+
+## Get a free token
+
+Please visit: https://www.puresec.io/function-shield-token-form
+
+### Modes
+
+- `'block'` - Block and log to Cloudwatch Logs
+- `'alert'` - Allow and log to Cloudwatch Logs
+- `'allow'` - Allow
+
+### Options
+
+
+- `policy.outbound_connectivity` - `'block'/'alert'/'allow'` (default: `'block'`)
+- `policy.read_write_tmp` - `'block'/'alert'/'allow'` (default: `'block'`)
+- `policy.create_child_process` - `'block'/'alert'/'allow'` (default: `'block'`)
+- `policy.read_handler` - `'block'/'alert'/'allow'` (default: `'block'`)
+- `token` - By default looks for `FUNCTION_SHIELD_TOKEN` in `process.env` and `context`
+- `disable_analytics` - Periodically, during cold starts, FunctionShield sends basic analytics information to its backend. To disable analytics module set: `true`. (default: `false`)
+
+### Sample Usage
+
+```javascript
+'use strict';
+
+const fs = require('fs');
+const child_process = require('child_process');
+const https = require('https');
+const middy = require('middy');
+const { ssm,  functionShield } = require('middy/middlewares');
+
+async function hello(event) {
+  try {
+    let fd = fs.openSync('/tmp/test', 'w');
+    fs.closeSync(fd);
+    console.log('successfully opened file in /tmp folder');
+  } catch (e) {
+    console.error('cannot open file in /tmp folder');
+  }
+  try {
+    child_process.execFileSync('/bin/uname');
+    console.log('successfully created child process');
+  } catch (e) {
+    console.error('cannot create child process');
+  }
+
+  let httpGet = (url) => {
+    return new Promise((resolve, reject) => {
+      https.get(url, (res) => {
+        resolve();
+      }).on('error', (e) => {
+        reject();
+      });
+    });
+  };
+  try {
+    await httpGet('https://nodejs.org');
+    console.log('successfully created outbound connection');
+  } catch (e) {
+    console.error('cannot create outbound connection');
+  }
+  try {
+    let fd = fs.openSync('/var/task/handler.js', 'r');
+    fs.closeSync(fd);
+    console.log('successfully opened handler file');
+  } catch (e) {
+    console.error('cannot open handler file');
+  }
+}
```

Comment:
I think this sample usage is quite confusing, looking at it I find myself asking "what is this sample code trying to show me?". I don't think it's necessary, and I just want to see the initialization bit below. The sample code should be minimal and shows me how to use this middleware, not what function shield does, which there should be explained in the description above, maybe with a link to a more in-depth article or the product landing page.